framework/include/functions/Function.h

```diff
@@ -111,55 +111,57 @@ class FunctionTempl : public MooseFunctionBase,
   void jacobianSetup() override;
 
 private:
-  using typename Moose::Functor<T>::FaceArg;
-  using typename Moose::Functor<T>::SingleSidedFaceArg;
-  using typename Moose::Functor<T>::ElemFromFaceArg;
-  using typename Moose::Functor<T>::ElemQpArg;
-  using typename Moose::Functor<T>::ElemSideQpArg;
-  using typename Moose::Functor<T>::ValueType;
-  using typename Moose::Functor<T>::GradientType;
-  using typename Moose::Functor<T>::DotType;
+  using typename Moose::FunctorImpl<T>::ValueType;
+  using typename Moose::FunctorImpl<T>::GradientType;
+  using typename Moose::FunctorImpl<T>::DotType;
 
   /**
    * @return the time associated with the requested \p state
    */
   Real getTime(unsigned int state) const;
 
-  ValueType evaluate(const Elem * const & elem, unsigned int state) const override final;
-  ValueType evaluate(const ElemFromFaceArg & elem_from_face,
+  ValueType evaluate(const Moose::ElemArg & elem, unsigned int state) const override final;
```

Comment:
Moose:: everywhere doesnt help readability.
let s add what we had back as `using typename Moose::ValueType;`

Reply:
This would be unnecessary if `Function` was in namespace `Moose`. this is one of the more horrible un-namespaced names